tokio-codec/src/framed_read.rs

```diff
@@ -211,7 +211,9 @@ where
             // Otherwise, try to read more data and try again. Make sure we've
             // got room for at least one byte to read to ensure that we don't
             // get a spurious 0 that looks like EOF
-            pinned.buffer.reserve(1);
+            if pinned.buffer.capacity() <= pinned.buffer.len() {
```

Comment:
`reserve` ensures that `capacity - length >= reserve_amount`. This means that `reserve(1)` is a no-op when `!pinned.buffer.is_empty()`. The additional guard is not needed. See [source](https://github.com/tokio-rs/bytes/blob/b5d4f87e7e6b7d9636aa42d3d3bd63503579b0b6/src/bytes.rs#L2307-L2315).

I would request reverting that commit.